### DISCUSSION TITLE
Tracking your progress on a larger time span.

### DISCUSSION THREAD

**Discussion by Brachyurus** [monkeytypegame/monkeytype#1355] [Ideas] [answered]
Basically, Right now I can check my typing stats of the tests done in the last 30 days or so, but I thought it would be cool if I could track my progress on a larger scale, for instance I could see how fast I could type 2 months ago and compare to my current results.

**Comment by unrealapex** [comment, marked as answer]
#1074

**Comment by Brachyurus** [comment]
So, to my understanding a maximum of 1000 tests can be saved for a user, but what if tests older than, for example two weeks were arranged into larger groups, then averaged into single values? This would mean that results wouldn't be as accurate, but users would be able to track their progress for longer periods of time without the need for extra storage space.

**Reply by unrealapex** [reply]
Could you clarify a bit more.

**Comment by Koushikphy** [comment]
@unrealapex Instead of storing information for all the typing test performed, for older days, it can just store the daily averages. Meaning, for a day 1 year back, I can see how many test I performed, time I took, average wpm, accuracy for that day etc. But I can not see in details each of the tests performed on that day. I particularly love how the way keybr shows the practice calendar.
